### ISSUE TITLE
Integration Details for DSEval Testing in TaskWeaver

### ISSUE THREAD

**Issue by Luffyzm3D2Y** [microsoft/TaskWeaver#446]
Hello TaskWeaver team,

I came across your paper mentioning TaskWeaver’s evaluation using the DSEval framework. I’m curious about how TaskWeaver was integrated with DSEval for testing. Could you kindly provide more details on the evaluation setup and any scripts you used during the testing process?

This information would be invaluable for replicating and understanding the evaluation process.

I browsed through the repository and found evaluation scripts for Eval-Cases, DS-1000, and DABench under the autoeval directory, but I couldn’t locate anything related to DSEval. Any pointers would be greatly appreciated!

Thank you very much for your time and assistance!

**Comment by ShilinHe** [comment]
Hi, @Luffyzm3D2Y, our evaluation is based on DSEval's code, and it might be too heavy to include the entire DSEval project into ours.
Regarding the implementation details, the main idea is: we hacked into DSEval by adding a new agent `TaskWeaver` to its agent.py file and reimplement its member functions such as extract_code. In this way, the python code generated by TaskWeaver CodeInterpreter can be evaluated in DSEval framework, just like existing ones (e.g., GPTAgent, JupyterAIAgent). We also made some minor modifications to other files like match.py, loop.py and simulation.py to make it better fit our settings.

**Comment by Luffyzm3D2Y** [comment]
@ShilinHe Thank you for your reply!

**Comment by Luffyzm3D2Y** [comment]
@ShilinHe I have another question. I am currently using the Taskweaver framework to run autoEval evaluations on DABench. Following the provided steps, a file named sample_case_results.csv is generated with the following format:
```
case_file, score, normalized_score
511, 1,  1.0
338, 2,  1.0
```
From my understanding, the case_file column records the ID of each problem. However, I am unclear about the meanings of the score and normalized_score columns. I would like to calculate the accuracy metrics for DABench (e.g., ABQ) and potentially other metrics. Could you please provide guidance on:
	* Accuracy Calculation: How can I compute the accuracy (such as ABQ) using the data in sample_case_results.csv?
	* Other Metrics: Is it possible to derive other accuracy-related metrics from this file? If so, how?
	* Detailed Results Access: How can I obtain more detailed results or metrics from DABench? I reviewed the documentation but couldn’t find an explanation of the sample_case_results.csv format. It’s possible I might have missed something.

Any clarification or additional documentation you could provide regarding the format of sample_case_results.csv and the calculation of evaluation metrics in DABench would be greatly appreciated.

**Comment by liqul** [comment]
Quick answer: If you have read our paper, one case can be transformed into a task with more than 1 steps. Each step has a certain score points, and different steps may have different score points. For example, you may consider step 1 is more important than step 2, so step 1 has a higher score point. So, in that csv file, the column score is the sum of all score points obtained for a case, which can be greater than 1. The normalized_score is normalized to 0~1. Hope this explains some of your doulbts.

**Comment by Luffyzm3D2Y** [comment]
Hello @liqul,

Thank you for clarifying the relationship between the score, normalized_score, and the metrics reported in your paper. So the accuracy given in the paper is the average normalized score calculated, which equals Accuracy Proportional by Sub-Question while in the InfiAgent Bench. I’m currently conducting a research project where I’d like to evaluate multiple metrics in this bench, not limited to only ABQ(which is defined in the Accuray in the InfiAgent Bench paper) or only QPSQ. 

Would it be possible for you to share the sample_case_results.csv files for GPT-3.5 and GPT-4 evaluated on InfiAgentBench by Taskweaver? Having access to these detailed outputs would enable me to compute additional metrics beyond those presented in the paper.  I would be delighted to cite your valuable paper in our work.

Thank you again for your assistance and support!

**Comment by liqul** [comment]
I don't think we have stored all the intermediate files, so not able to share with you.

**Comment by Luffyzm3D2Y** [comment]
@liqul Got it! Thank you for letting me know. I appreciate your response, and thanks again for your help!
